packages/react-next/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -308,142 +308,194 @@ const hiddenDivStyles: React.CSSProperties = { position: 'fixed', visibility: 'h
-  }
+  // A div that can be used for the initial measurement so that we can avoid mounting a second instance
+  // of the component being measured for the initial render.
+  const initialHiddenDiv = React.useRef<HTMLDivElement | null>(null);
+  // A hidden div that is used for mounting a new instance of the component for measurement in a hidden
+  // div without unmounting the currently visible content.
+  const updateHiddenDiv = React.useRef<HTMLDivElement | null>(null);
 
-  public componentWillUnmount(): void {
-    this._async.dispose();
-    this._events.dispose();
-  }
+  // Tracks if any content has been rendered to the user. This enables us to do some performance optimizations
+  // for the initial render.
+  const hasRenderedContent = React.useRef(false);
+
+  const async: Async = useAsync();
 
-  public remeasure(): void {
-    if (this._root.current) {
-      this.setState({ measureContainer: true });
+  const [stateRef, updateResizeState, remeasure] = useResizeState(props, nextResizeGroupStateProvider, rootRef);
+
+  React.useEffect(() => {
+    if (stateRef.current.renderedData) {
+      hasRenderedContent.current = true;
+      props.dataDidRender?.(stateRef.current.renderedData);
     }
-  }
+  });
 
-  private _afterComponentRendered(direction?: ResizeGroupDirection): void {
-    this._async.requestAnimationFrame(() => {
+  React.useEffect((): void => {
+    async.requestAnimationFrame(() => {
       let containerDimension = undefined;
-      if (this.state.measureContainer && this._root.current) {
-        const boundingRect = this._root.current.getBoundingClientRect();
+      if (stateRef.current.measureContainer && rootRef.current) {
+        const boundingRect = rootRef.current.getBoundingClientRect();
         containerDimension =
-          direction && direction === ResizeGroupDirection.vertical ? boundingRect.height : boundingRect.width;
+          props.direction === ResizeGroupDirection.vertical ? boundingRect.height : boundingRect.width;
       }
-      const nextState = this._nextResizeGroupStateProvider.getNextState(
-        this.props,
-        this.state,
+      const nextState = nextResizeGroupStateProvider.getNextState(
+        props,
+        stateRef.current,
         () => {
-          const refToMeasure = !this._hasRenderedContent ? this._initialHiddenDiv : this._updateHiddenDiv;
+          const refToMeasure = !hasRenderedContent.current ? initialHiddenDiv : updateHiddenDiv;
           if (!refToMeasure.current) {
             return 0;
           }
-          return direction && direction === ResizeGroupDirection.vertical
+          return props.direction === ResizeGroupDirection.vertical
             ? refToMeasure.current.scrollHeight
             : refToMeasure.current.scrollWidth;
         },
         containerDimension,
       );
 
-      if (nextState) {
-        this.setState(nextState);
-      }
+      updateResizeState(nextState);
     });
-  }
+  });
```

Comment:
Again no deps, and also I see no symmetrical dispose here.

Reply:
What would there be to dispose? The `async` instance is disposed in `useAsync`.